src/util/color.ts

```diff
@@ -1,27 +1,29 @@
-import {parseCSSColor} from 'csscolorparser';
+import colorString from 'color-string';
+import {hcl, hsl, lab, HCLColor, LABColor, RGBColor} from './color_spaces';
 
 /**
- * An RGBA color value. Create instances from color strings using the static
- * method `Color.parse`. The constructor accepts RGB channel values in the range
- * `[0, 1]`, premultiplied by A.
- *
- * @param {number} r The red channel.
- * @param {number} g The green channel.
- * @param {number} b The blue channel.
- * @param {number} a The alpha channel.
+ * Color representation used by WebGL.
+ * Defined in sRGB color space and pre-blended with alpha.
  * @private
  */
 class Color {
-    r: number;
-    g: number;
-    b: number;
-    a: number;
 
-    constructor(r: number, g: number, b: number, a: number = 1) {
+    readonly r: number;
```

Comment:
This can be declared right in the constructor, no need to repeat these 4 variables.

Reply:
I changed this constructor to make the colors created by the `rgb` and `rgba` expressions behave the same as those created from CSS string parsing. So it's no longer relevant, I think.

Reply:
I mean you can write `constructor(readonly a: number...` instead of defining the members and then assigning them in the constructor.

Reply:
Personally, I find the TS syntax of defining properties of a class in constructor hard to read.
I don't see a problem with their repetition, I treat it as much as defining an interface as some kind of documentation.
It is easier to identify the properties of a class when they are defined in this way
https://github.com/maplibre/maplibre-style/blob/e679ba7e2bf31c5d7ed99110bae654c5d257beb2/src/util/color.ts#L9-L15
rather than looking for arguments preceded by `public` `readonly` etc. in its constructor.
I will leave it as it is now, if you feel strongly about it you can change it later.